text/0000-safer-transmute.md

```diff
@@ -643,93 +532,73 @@ Even with `NeglectValidity`, the compiler will statically reject transmutations
 
-        /* A, B and !::C are equal */
-        assert_type_eq_all!(A, B, !::C);
-
-        /* C and D are NOT equal; C and D inhabit different scopes */
-        assert_type_ne_all!(C, D);
-    };
+You might also use this option to signal that a *particular* transmutation is stable *without* implementing `Muckable` (which would signal that *all* transmutations are stable):
+```rust
+impl From<Foo> for Bar
+where
+    Bar: TransmuteFrom<Foo, NeglectSafety>
+{
+    fn from(src: Foo) -> Self {
+        unsafe { Bar::unsafe_transmute_from(src) }
+    }
 }
 ```
-Scope types should (as much as possible) pretty print in compiler error messages as their definition path.
 
-These scope types should generated with `pub(self)` visibility. We are not currently aware of any reason why publicly re-exporting a scope type via a type alias would be a good idea; restricting the visibility of these types will warn users against doing so. If compelling use-cases for re-exported scope types are discovered in the future, a broader visibility could be used instead without breaking backwards compatibility.
-
-### Implementing `Constructible`
-The compiler implements `Constructible<Scope>` for `T` if `T` is fully implicitly constructible in the scope uniquely identified by the type `Scope`.
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
 
-A type `T` is fully implicitly constructible in a particular scope if:
-  - `T`'s implicit constructor is reachable from the scope, and either:
-    - `T` has no fields, or
-    - `T`'s fields are fully implicitly constructible from the scope.
+## Implementation Guidance
+Two items in this RFC require special compiler support:
+  - `Muckable`
+  - `TransmuteFrom`
+  - `differing_sizes` lint
 
-The `Constructible` trait does not ever need to be made stable, or even visible (à la the virtual `Freeze` trait). It is merely a useful device for implementing `TransmuteFrom`.
+### Implementing `Muckable`
+The `Muckable` marker trait is similar to `Copy` in that all fields of a `Muckable` type must, themselves, be `Muckable`.
 
 ### Implementing `TransmuteFrom`
 The implementation of `TransmuteFrom` is completely internal to the compiler (à la [`Sized`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyS.html#method.is_sized) and [`Freeze`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyS.html#method.is_freeze)).
 
 #### Constructability and Transmutation
-Unless `NeglectConstructability` is used as `Neglect` option, a `Src` is *safely* transmutable into `Dst` in a given `Scope` if:
-  1. `Dst: Constructible<Scope>`
-  2. `Dst: TransmuteFrom<Src, Scope, Neglect>`
+A `Src` is *safely* transmutable into `Dst` in a given if:
+  1. `Dst: Muckable`
+  2. `Dst: TransmuteFrom<Src, Neglect>`
+  3. `NeglectSafety` ∉ `Neglect`
 
 If `Src` is a mutatable reference, then additionally:
```

Comment:
"is or contains", I assume?

Reply:
> "is or contains", I assume?

You assume correctly. I'll just use "is or contains" instead.